grpc/src/main/java/io/stargate/grpc/service/SchemaNotificationsHandler.java

```diff
@@ -43,33 +47,63 @@
   }
 
   public void handle() {
+    synchronized (responseObserver) {
+      responseObserver.setOnCancelHandler(() -> persistence.unregisterEventListener(this));
+      responseObserver.setOnCloseHandler(() -> persistence.unregisterEventListener(this));
+    }
```

Comment:
For non-streaming handlers I think we're fine because observer methods will only be invoked from a single thread.

Whether we synchronize on `this` or `responseObserver` does not matter, as long as it's done consistently. But generally I prefer using a private field, this avoids issues if a caller also happens to synchronize on this object and deadlocks (granted the risk is low here).